osmoutils/accum/accum.go

```diff
@@ -63,14 +63,30 @@ func setAccumulator(accum AccumulatorObject, amt sdk.DecCoins) {
 	osmoutils.MustSet(accum.store, formatAccumPrefixKey(accum.name), &newAccum)
 }
 
-// UpdateAccumulator updates the accumulator's value by amt.
-// It does so by incresing the value of the accumulator by
+// AddToAccumulator updates the accumulator's value by amt.
+// It does so by increasing the value of the accumulator by
 // the given amount. Persists to store. Mutates the receiver.
-func (accum *AccumulatorObject) UpdateAccumulator(amt sdk.DecCoins) {
+// make this safely return error when negative dec coins
+func (accum *AccumulatorObject) AddToAccumulator(amt sdk.DecCoins) {
 	accum.value = accum.value.Add(amt...)
 	setAccumulator(*accum, accum.value)
 }
 
+// SubFromAccumulator subtracts the given amt from the fee accumulator.
+// Safesub is used to panic and if given amt is greater than the value of accumulator,
+// returns error.
+func (accum *AccumulatorObject) SubFromAccumulator(amt sdk.DecCoins) error {
```

Comment:
We should be never subtracting from an accumulator. It is always increasing by-design

Reply:
resolved